// javascript-decorators/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
         value: true
});

var _immutablors = require('./@immutablors.js');

Object.defineProperty(exports, 'immutable', {
         enumerable: true,
         get: function get() {
                  return _immutablors._immutable;
         }
});
Object.defineProperty(exports, 'doesNotMutate', {
         enumerable: true,
         get: function get() {
                  return _immutablors._doesNotMutate;
         }
});

var _validators = require('./@validators.js');

Object.defineProperty(exports, 'acceptsArray', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsArray;
         }
});
Object.defineProperty(exports, 'acceptsObject', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsObject;
         }
});
Object.defineProperty(exports, 'acceptsInteger', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsInteger;
         }
});
Object.defineProperty(exports, 'acceptsNumber', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsNumber;
         }
});
Object.defineProperty(exports, 'acceptsBoolean', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsBoolean;
         }
});
Object.defineProperty(exports, 'acceptsFunction', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsFunction;
         }
});
Object.defineProperty(exports, 'acceptsPromise', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsPromise;
         }
});
Object.defineProperty(exports, 'acceptsString', {
         enumerable: true,
         get: function get() {
                  return _validators._acceptsString;
         }
});
Object.defineProperty(exports, 'validateSchema', {
         enumerable: true,
         get: function get() {
                  return _validators._validateSchema;
         }
});

var _memoizator = require('./@memoizator');

Object.defineProperty(exports, 'memoization', {
         enumerable: true,
         get: function get() {
                  return _memoizator._memoization;
         }
});

var _timers = require('./@timers');

Object.defineProperty(exports, 'debounce', {
         enumerable: true,
         get: function get() {
                  return _timers._debounce;
         }
});
Object.defineProperty(exports, 'timeout', {
         enumerable: true,
         get: function get() {
                  return _timers._timeout;
         }
});
Object.defineProperty(exports, 'defer', {
         enumerable: true,
         get: function get() {
                  return _timers._defer;
         }
});

var _loggers = require('./@loggers');

Object.defineProperty(exports, 'loglocalstorage', {
         enumerable: true,
         get: function get() {
                  return _loggers._loglocalstorage;
         }
});
Object.defineProperty(exports, 'log', {
         enumerable: true,
         get: function get() {
                  return _loggers._log;
         }
});
Object.defineProperty(exports, 'donotlog', {
         enumerable: true,
         get: function get() {
                  return _loggers._donotlog;
         }
});
Object.defineProperty(exports, 'donotlogmessages', {
         enumerable: true,
         get: function get() {
                  return _loggers._donotlogmessages;
         }
});
Object.defineProperty(exports, 'donotlogerrors', {
         enumerable: true,
         get: function get() {
                  return _loggers._donotlogerrors;
         }
});
Object.defineProperty(exports, 'donotlogwarnings', {
         enumerable: true,
         get: function get() {
                  return _loggers._donotlogwarnings;
         }
});

var _executors = require('./@executors');

Object.defineProperty(exports, 'once', {
         enumerable: true,
         get: function get() {
                  return _executors._once;
         }
});
Object.defineProperty(exports, 'times', {
         enumerable: true,
         get: function get() {
                  return _executors._times;
         }
});
Object.defineProperty(exports, 'timesCalled', {
         enumerable: true,
         get: function get() {
                  return _executors._timesCalled;
         }
});

var _stators = require('./@stators');

Object.defineProperty(exports, 'readonly', {
         enumerable: true,
         get: function get() {
                  return _stators._readonly;
         }
});
Object.defineProperty(exports, 'enumerable', {
         enumerable: true,
         get: function get() {
                  return _stators._enumerable;
         }
});
Object.defineProperty(exports, 'nonenumerable', {
         enumerable: true,
         get: function get() {
                  return _stators._nonenumerable;
         }
});
Object.defineProperty(exports, 'nonconfigurable', {
         enumerable: true,
         get: function get() {
                  return _stators._nonconfigurable;
         }
});

var _inheritedfunctions = require('./@inheritedfunctions');

Object.defineProperty(exports, 'overridden', {
         enumerable: true,
         get: function get() {
                  return _inheritedfunctions._overridden;
         }
});
Object.defineProperty(exports, 'forceoverridden', {
         enumerable: true,
         get: function get() {
                  return _inheritedfunctions._forceoverridden;
         }
});

var _trycatch2 = require('./@trycatch');

Object.defineProperty(exports, 'trycatch', {
         enumerable: true,
         get: function get() {
                  return _trycatch2._trycatch;
         }
});

var _multiinheritance = require('./@multiinheritance');

Object.defineProperty(exports, 'multiInherit', {
         enumerable: true,
         get: function get() {
                  return _multiinheritance._multiInherit;
         }
});
Object.defineProperty(exports, 'multiExtend', {
         enumerable: true,
         get: function get() {
                  return _multiinheritance._multiInherit;
         }
});
Object.defineProperty(exports, 'partialyInherit', {
         enumerable: true,
         get: function get() {
                  return _multiinheritance._partialyInherit;
         }
});
Object.defineProperty(exports, 'partialyExtend', {
         enumerable: true,
         get: function get() {
                  return _multiinheritance._partialyInherit;
         }
});
Object.defineProperty(exports, 'partiallyInherit', {
         enumerable: true,
         get: function get() {
                  return _multiinheritance._partialyInherit;
         }
});
Object.defineProperty(exports, 'partiallyExtend', {
         enumerable: true,
         get: function get() {
                  return _multiinheritance._partialyInherit;
         }
});

var _passedValuesEqualToNumberOfArguments2 = require('./@passedValuesEqualToNumberOfArguments');

Object.defineProperty(exports, 'passedValuesEqualToNumberOfArguments', {
         enumerable: true,
         get: function get() {
                  return _passedValuesEqualToNumberOfArguments2._passedValuesEqualToNumberOfArguments;
         }
});
Object.defineProperty(exports, 'valuesEqualToNumberOfArguments', {
         enumerable: true,
         get: function get() {
                  return _passedValuesEqualToNumberOfArguments2._passedValuesEqualToNumberOfArguments;
         }
});

var _after2 = require('./@after');

Object.defineProperty(exports, 'after', {
         enumerable: true,
         get: function get() {
                  return _after2._after;
         }
});

var _before2 = require('./@before');

Object.defineProperty(exports, 'before', {
         enumerable: true,
         get: function get() {
                  return _before2._before;
         }
});

var _deprecated2 = require('./@deprecated');

Object.defineProperty(exports, 'deprecated', {
         enumerable: true,
         get: function get() {
                  return _deprecated2._deprecated;
         }
});
Object.defineProperty(exports, 'deprecate', {
         enumerable: true,
         get: function get() {
                  return _deprecated2._deprecated;
         }
});

var _compose2 = require('./@compose');

Object.defineProperty(exports, 'compose', {
         enumerable: true,
         get: function get() {
                  return _compose2._compose;
         }
});
Object.defineProperty(exports, 'rightCompose', {
         enumerable: true,
         get: function get() {
                  return _compose2._compose;
         }
});
Object.defineProperty(exports, 'leftCompose', {
         enumerable: true,
         get: function get() {
                  return _compose2._leftCompose;
         }
});

var _autobind2 = require('./@autobind');

Object.defineProperty(exports, 'autobind', {
         enumerable: true,
         get: function get() {
                  return _autobind2._autobind;
         }
});

var _abstract2 = require('./@abstract');

Object.defineProperty(exports, 'abstract', {
         enumerable: true,
         get: function get() {
                  return _abstract2._abstract;
         }
});
